Plots/Overlays/NCL_overlay_11a.py

```diff
@@ -0,0 +1,213 @@
+"""
+NCL_overlay_11a.py
+==================
+
+Concepts illustrated:
+
+- Overlaying vectors and filled contours on a map
+- Masking out particular areas in a map
+- Subsetting a color map
+
+This script shows how to overlay contours and vectors on a map,
+but with the contours limited to specific areas, and the vectors
+not limited.
+
+The point of this script is to show how to mask contours against
+a geographical boundary, but in a way that allows them to be drawn
+up to the boundary location. This is unlike the shapefile masking
+examples, where grid points are set to missing if they fall
+outside a boundary, and hence you can get blocky features close
+to the boundary.
+
+With Python's matplotlib, there are 2 general approaches to
+accomplishing this:
+
+a. You can "cover" (i.e., "over lay") geographical features on
+   top of other plots using the ``zorder`` parameter to most
+   rendered objects.
+
+b. You can "clip" a plot object with a geographical boundary.
+
+This example demonstrates approach (a).
+
+This script is based on the NCL script originally written by
+Yang Zhao (CAMS) (Chinese Academy of Meteorological Sciences).
+"""
+
+###############################################################################
+# Basic Imports
+# -------------
+
+import xarray as xr
+import numpy as np
+import cmaps
+
+from shapely.geometry import MultiPolygon
+
+from matplotlib import pyplot as plt
+from matplotlib.colors import LinearSegmentedColormap
+from matplotlib.ticker import AutoMinorLocator
+from matplotlib.patches import PathPatch
+
+from cartopy.feature import ShapelyFeature, OCEAN, LAKES, LAND
+from cartopy.crs import PlateCarree
+from cartopy.mpl.patch import geos_to_path
+from cartopy.io.shapereader import Reader as ShapeReader, natural_earth
+from cartopy.mpl.ticker import LongitudeFormatter, LatitudeFormatter
+
+###############################################################################
+# Read U,V,T from the data at 500hPa
+# ----------------------------------
+#
+# Here, we read the sample dataset with Xarray and select the ``time=0`` slice
+# and the ``lev=500`` hPa level.
+
+ds = xr.open_dataset('../../data/netcdf_files/uvt.nc').sel(time=0, lev=500)
+
+# For convenience only, extract the U,V,T and lat and lon variables
+U = ds["U"]
+V = ds["V"]
+T = ds["T"]
+
+lat = ds["lat"]
+lon = ds["lon"]
+
+# Define the contour levels (T) we are interested in plotting
+clevs = np.arange(228, 273, 4, dtype=float)
+
+###############################################################################
+# Create a subselection of the color map
+# --------------------------------------
+#
+# We create a new color map that is a subselection of an existing color map
+
+trunc_cm = LinearSegmentedColormap.from_list(
+    name="BkBlAqGrYeOrReViWh200",
+    colors=cmaps.BkBlAqGrYeOrReViWh200(np.linspace(0.1, 0.6, len(clevs))),
+)
+plt.cm.register_cmap('BkBlAqGrYeOrReViWh200', trunc_cm)
```

Comment:
This can be done with `geocat.viz`'s `truncate_colormap()` function:

```python
import geocat.viz as gcv

cmap = gcv.util.truncate_colormap(cmaps.BkBlAqGrYeOrReViWh200, 0.1, 0.6, len(clevs), "BkBlAqGrYeOrReViWh200"
```

and I think it's a good idea to advertise `gcv` as much as possible

Reply:
This is excellent!  I'll give that a shot.

Reply:
Done.